metricflow/query/query_parser.py

```diff
@@ -172,6 +176,65 @@ def _top_fuzzy_matches(
         top_ranked_suggestions = [x for x in top_ranked_suggestions if x[1] > min_score]
         return [x[0] for x in top_ranked_suggestions]
 
+    def parse_and_validate_saved_query(
+        self,
+        saved_query_parameter: SavedQueryParameter,
+        where_filter: Optional[WhereFilter],
+        limit: Optional[int],
+        order_by_parameters: Optional[Sequence[OrderByQueryParameter]],
+    ) -> MetricFlowQuerySpec:
+        """Parse and validate a query using parameters from a pre-defined / saved query.
+
+        Additional parameters act in conjunction with the parameters in the saved query.
+        """
+        saved_query = self._get_saved_query(saved_query_parameter)
+
+        # This logic could be encapsulated in the WhereFilter through a merge interface.
```

Comment:
Yes, and even more so now that the object is going to hold a list of WhereFilters.

I haven't added it yet, but it might be something we should add in to the base protocol. If we don't think that's worth doing we can re-type everything internally to take an MFWhereFilter that just extends the WhereFilter protocol with merge or combine or whatever.

Reply:
Yeah, we could add `Mergeable` to the base class, but we can visit that once `Mergeable` is in.